src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,123 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_LENGTH = 10;
+    private T[] data;
+    private int size;
+
+    public ArrayList() {
+        data = (T[]) new Object[DEFAULT_LENGTH];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        arrayIncrease(size + 1);
```

Comment:
I would say that is a bit confusing when we are calling arrayIncrease each time. Suggest to call it only, if internal array is full and make this condition explicit.